Error at generating PlatformIO manifest

Hi There

The ./build.sh script ends with:
> fatal: ref HEAD is not a symbolic ref
> usage: gen_platformio_manifest.py [-h] -o DST_DIR -s VERSION_STRING -c COMMIT_HASH
> gen_platformio_manifest.py: error: argument -s/--version-string: expected one argument

It seems to me, however, related to the patches done to esp_eth component.

> $ git symbolic-ref --short HEAD
> fatal: ref HEAD is not a symbolic ref

and:

> $ git status
> HEAD detached at cb174b0fe1
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         modified:   components/esp_eth/include/esp_eth_mac.h
>         modified:   components/esp_eth/src/esp_eth_mac_dm9051.c
>         modified:   components/esp_eth/src/esp_eth_mac_ksz8851snl.c
>         modified:   components/esp_eth/src/esp_eth_mac_w5500.c
> 
> no changes added to commit (use "git add" and/or "git commit -a")


I've temporarily override it by a manual generation.

@me-no-dev Having a similar issue as discussed above, except running `./build.sh` fails at:

```
-- Generating done (1.2s)
-- Build files have been written to: /Users/brentrubell/Desktop/github_brentru/esp32-arduino-lib-builder/build
make: *** No rule to make target `idf_libs'.  Stop.
command "idf_libs" is not known to idf.py and is not a Unix Makefiles target
```

I also think it may be due to the patches within `esp_eth` as they require a manual confirmation prompt the second time you run the script. 

I've included the script output on this gist and highlighted the line with the symbolic ref issue: https://gist.github.com/brentru/aa9acf05b808575b753f6e461509bf35#file-gistfile1-txt-L281

Everything is configured on a local machine running macOS and `bash` shell.

guys appreciated your work and everything but the workflow seems not clear to me, iam an idiot and iam desperate.

i forked it and did the GA thing, now for testing without my modifications.
The GA got me the 3 files as posted in the screenshot before.
Jason you told me i should download the artefacts file and i did. (i only downloaded artefacts and not the other two)
You told me the folder structure does not differ but it does.
the most matching folders i found:

now on windows:
C:\Users\user\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11\tools\

**there is a folder sdk containing**
![grafik](https://github.com/espressif/esp32-arduino-lib-builder/assets/142450769/c563e876-bcc8-4d7c-944c-37c75d15b565)

the artefacts folder has the folder tools\esp32-arduino-libs\
![grafik](https://github.com/espressif/esp32-arduino-lib-builder/assets/142450769/779f3f05-8194-4375-a721-21e6386829ae)

so i copied and overwrote all files from the artefacts\tools\esp32-arduino-libs\ to C:\Users\user\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11\tools\sdk\

(didnt delete the old folders. i tried that too and got even more errors like not finding freertos.h etc.)

then i restartet arduino ide 2.1.1, tried to compile and i got errors.

```
C:\Users\user\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11/tools/sdk/esp32s3/include/driver/include/driver/gpio.h:71:29: error: 'gpio_config_t' does not name a type; did you mean 'gpio_mode_t'?
 esp_err_t gpio_config(const gpio_config_t *pGPIOConfig);
```

```
C:\Users\sk3\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11/tools/sdk/esp32s3/include/driver/include/driver/gpio.h:326:53: error: 'gpio_isr_t' has not been declared
 esp_err_t gpio_isr_handler_add(gpio_num_t gpio_num, gpio_isr_t isr_handler, void *args);
```

am i missing a simple step which is clear for you pro's but not for me?

kind regards

--- a/build.sh
+++ b/build.sh
@@ -276,7 +276,11 @@ fi
 
 # Generate PlatformIO manifest file
 if [ "$BUILD_TYPE" = "all" ]; then
-    python3 ./tools/gen_platformio_manifest.py -o "$TOOLS_JSON_OUT/" -s $(git -C "$IDF_PATH" symbolic-ref --short HEAD || git -C "$IDF_PATH" tag --points-at HEAD) -c $(git -C "$IDF_PATH" rev-parse --short HEAD)
+    pushd $IDF_PATH
+    ibr=$(git describe --all --exact-match 2>/dev/null)
+    ic=$(git -C "$IDF_PATH" rev-parse --short HEAD)
+    popd
+    python3 ./tools/gen_platformio_manifest.py -o "$TOOLS_JSON_OUT/" -s "$ibr" -c "$ic"
     if [ $? -ne 0 ]; then exit 1; fi
 fi
 

